bundles/org.openhab.binding.teleinfo/src/main/java/org/openhab/binding/teleinfo/internal/serial/TeleinfoReceiveThread.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.teleinfo.internal.serial;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.io.transport.serial.SerialPort;
+import org.openhab.binding.teleinfo.internal.dto.Frame;
+import org.openhab.binding.teleinfo.internal.reader.io.TeleinfoInputStream;
+import org.openhab.binding.teleinfo.internal.reader.io.serialport.InvalidFrameException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TeleinfoReceiveThread} class defines a thread to decode and fire Teleinfo frames for Serial controller.
+ *
+ * @author Nicolas SIBERIL - Initial contribution
+ */
+@NonNullByDefault
+public class TeleinfoReceiveThread extends Thread {
+
+    private final Logger logger = LoggerFactory.getLogger(TeleinfoReceiveThread.class);
+
+    private SerialPort serialPort;
+    private @Nullable TeleinfoReceiveThreadListener listener;
+    private boolean autoRepairInvalidADPSgroupLine;
+
+    public TeleinfoReceiveThread(SerialPort serialPort, final TeleinfoReceiveThreadListener listener,
+            boolean autoRepairInvalidADPSgroupLine) {
+        super("TeleinfoReceiveThread");
+
```

Comment:
Created threads should be daemon.
```suggestion
        setDaemon(true);
```